akka-actor/src/main/scala/akka/actor/Scheduler.scala

```diff
@@ -70,51 +71,25 @@ trait Scheduler {
    * Note: For scheduling within actors `with Timers` should be preferred.
    */
   def scheduleWithFixedDelay(initialDelay: FiniteDuration, delay: FiniteDuration)(runnable: Runnable)(
-      implicit executor: ExecutionContext): Cancellable = {
-    try new AtomicReference[Cancellable](Cancellable.initialNotCancelled) with Cancellable { self =>
-      compareAndSet(
-        Cancellable.initialNotCancelled,
+      implicit executor: ExecutionContext): Cancellable =
+    new AtomicCancellable(Cancellable.initialNotCancelled) {
+      final override protected def next(): Cancellable =
         scheduleOnce(
           initialDelay,
           new Runnable {
             override def run(): Unit = {
               try {
                 runnable.run()
-                if (self.get != null)
+                if (get != null)
```

Comment:
in the other place it is
```
if (self.get() != null)
```

Reply:
why is `self =>` used there but not here?

Reply:
Because I need to access the `AtomicReference` there but not the outer `AtomicLong`.